db/migrate/20170512042542_create_categories.rb

```diff
@@ -0,0 +1,15 @@
+class CreateCategories < ActiveRecord::Migration
+  def change
+    create_table :categories do |t|
+      t.string :canonical_title, required: true
+      t.integer :anidb_id, index: true
+      t.string :image_content_type, limit: 255
+      t.string :image_file_name, limit: 255
+      t.integer :image_file_size
+      t.datetime :image_updated_at
```

Comment:
Yes, you will be able to assign it to the virtual attribute.

```ruby
object.image = <image_path>
```

Upon saving, it'll grab the image and fill in the related fields.